libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -644,14 +643,27 @@ template <typename ServerType>
 void ESP8266WebServerTemplate<ServerType>::collectHeaders(const char* headerKeys[], const size_t headerKeysCount) {
   _headerKeysCount = headerKeysCount + 1;
   if (_currentHeaders)
-     delete[]_currentHeaders;
+    delete[] _currentHeaders;
   _currentHeaders = new RequestArgument[_headerKeysCount];
   _currentHeaders[0].key = FPSTR(AUTHORIZATION_HEADER);
   for (int i = 1; i < _headerKeysCount; i++){
     _currentHeaders[i].key = headerKeys[i-1];
   }
 }
 
+template <typename ServerType>
```

Comment:
It's been a while since I last saw this use of variadic args. Pretty nice!
There are several things at play here, but most should be ok for 99% of use cases, so it's ok. 
However, I don't like the code duplication. This method body is almost the same as the method above it, and I was already confused for a moment due to the different +1.
How about implementing this method in terms of the above, variadic args and all? Something along the lines of declaring an array of headers with args, and then call the above method?

Reply:
> However, I don't like the code duplication.

Who likes code duplication? :)

> How about implementing this method in terms of the above, variadic args and all? Something along the lines of declaring an array of headers with args, and then call the above method?

What you wanna say I guess is:
```
// [ ESP8266WebServer.h ]
protected:
  void _collectHeaders(String headerKeys[], const size_t headerKeysCount);
```
```
// [ ESP8266WebServer-impl.h ]
template <typename ServerType>
void ESP8266WebServerTemplate<ServerType>::_collectHeaders(String headerKeys[], const size_t headerKeysCount) {
  _headerKeysCount = headerKeysCount + 1;
  if (_currentHeaders)
    delete[] _currentHeaders;
  _currentHeaders = new RequestArgument[_headerKeysCount];
  _currentHeaders[0].key = FPSTR(AUTHORIZATION_HEADER);
  for (int i = 1; i < _headerKeysCount; i++){
    _currentHeaders[i].key = headerKeys[i-1];
  }
}

template <typename ServerType>
void ESP8266WebServerTemplate<ServerType>::collectHeaders(const char* headerKeys[], const size_t headerKeysCount) {
  String headers[headerKeysCount];
  for (int i = 0; i < headerKeysCount; i++) {
    headers[i] = String(headerKeys[i]);
  }
  _collectHeaders(headers, headerKeysCount);
}

template <typename ServerType>
template <typename... Args>
void ESP8266WebServerTemplate<ServerType>::collectHeaders(const Args&... args) {
  String headers[] = {
    args...
  };
  _collectHeaders(headers, sizeof...(args));
}
```
and inspection code:
```
#include <ESP8266WebServer.h>
static ESP8266WebServer webServer(80);
void setup() {
  webServer.collectHeaders(F("Content-Type"), F("Origin"));
}
void loop() {
  webServer.handleClient();
}
```
Finally, IROM grows 251072 bytes to 251152... inefficient, vain efforts.